evmcore/tx_validation.go

```diff
@@ -125,39 +118,76 @@ func validateTx(tx *types.Transaction, opt validationOptions) error {
 		return ErrTipAboveFeeCap
 	}
 
+	// For now, Sonic only supports Blob transactions without blob data.
+	if tx.BlobHashes() != nil && (len(tx.BlobHashes()) > 0 ||
+		(tx.BlobTxSidecar() != nil && len(tx.BlobTxSidecar().BlobHashes()) > 0)) {
+		return ErrTxTypeNotSupported
```

Comment:
Given the granularity of other error types, this situation should get its own error definition instead of re-using the "ErrTxTypeNotSupported".

Reply:
https://github.com/0xsoniclabs/sonic/pull/200/commits/2724ab4e13337c83cbe9efe010de5df4c21bf2ba